functional_tests/test_images.py

```diff
@@ -0,0 +1,138 @@
+
+    def test_images_pull(self):
+        _, out, _ = run_doozer([
+            "--group=openshift-3.11",
+            "--images=aos3-installation",
+            "images:pull",
+        ])
+
+    def test_images_show_tree(self):
+        _, out, _ = run_doozer([
+            "--group=openshift-3.11",
+            "--images=grafana,openshift-enterprise-cli,openshift-enterprise-base",
+            "images:show-tree",
+            "--yml"
+        ])
+        tree = yaml.safe_load(out)
+        self.assertIn("grafana", tree["openshift-enterprise-base"])
+        self.assertIn("openshift-enterprise-cli", tree["openshift-enterprise-base"])
+
+    def test_images_push(self):
+        _, _, err = run_doozer([
+            "--group=openshift-3.11",
+            "--images=aos3-installation",
+            "images:push",
+            "--to-defaults",
+            "--dry-run",
+        ])
+        self.assertIn("Would have tagged", err)
+
+    def test_images_rebase(self):
+        _, out, _ = run_doozer([
+            "--group=openshift-3.11",
+            "--images=aos3-installation",
+            "images:rebase",
+            "--message=test rebase"
+        ])
+        working_dir = get_working_dir()
+        repo_path = os.path.join(working_dir, "distgits", "containers", "aos3-installation")
+        repo = pygit2.Repository(repo_path)
+        ref = repo.references["HEAD"].resolve()  # type: pygit2.Reference
+        git_commit = ref.peel(pygit2.Commit)  # type: pygit2.Commit
+        self.assertIn("test rebase", git_commit.message)
+
+    def test_images_update_dockerfile(self):
+        _, out, _ = run_doozer([
+            "--group=openshift-3.11",
+            "--images=aos3-installation",
+            "images:update-dockerfile",
+            "--message=test update dockerfile"
+        ])
+        working_dir = get_working_dir()
+        repo_path = os.path.join(working_dir, "distgits", "containers", "aos3-installation")
+        repo = pygit2.Repository(repo_path)
+        ref = repo.references["HEAD"].resolve()  # type: pygit2.Reference
+        git_commit = ref.peel(pygit2.Commit)  # type: pygit2.Commit
+        self.assertIn("test update dockerfile", git_commit.message)
+
+    def test_images_revert(self):
+        _, out, _ = run_doozer([
+            "--group=openshift-3.11",
+            "--images=aos3-installation",
+            "images:revert",
+            "--message=test revert",
+            "1",
+        ])
+        working_dir = get_working_dir()
+        repo_path = os.path.join(working_dir, "distgits", "containers", "aos3-installation")
+        repo = pygit2.Repository(repo_path)
+        ref = repo.references["HEAD"].resolve()  # type: pygit2.Reference
+        git_commit = ref.peel(pygit2.Commit)  # type: pygit2.Commit
+        self.assertIn("test revert", git_commit.message)
+
+    def test_images_push_distgit(self):
+        # FIXME: images:push-distgit is not tested because we don't have an option to dry run
```

Comment:
we could push to a private branch (any branch beginning with `private-` which is allowed to create on demand) and then nuke the branch afterward.